src/main/java/com/iabtcf/decoder/ConsentParser.java

```diff
@@ -0,0 +1,311 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import java.time.Instant;
+import java.util.List;
+
+import com.iabtcf.utils.IntIterable;
+import com.iabtcf.v2.PublisherRestriction;
+
+public interface ConsentParser {
+    public static ConsentParser parse(String consentString) {
+        return ConsentDecoder.parse(consentString);
+    }
+
+    /**
+     * Version number of the encoding format
+     * 
+     * @since 1.0
+     * @return the version number
+     */
+    byte getVersion();
+
+    /**
+     * Epoch deciseconds (0.1 of a second) when this TC String was first created
+     *
+     * @since 1.0
+     * @return timestamp the record was first created
+     */
+    Instant getCreated();
+
+    /**
+     * Epoch deciseconds (0.1 of a second) when TC String was last updated
+     *
+     * @since 1.0
+     * @return timestamp record was last updated
+     */
+    Instant getLastUpdated();
+
+    /**
+     * Consent Management Platform ID that last updated the TC String
+     *
+     * @since 1.0
+     * @return the Consent Management Platform ID
+     */
+    short getCmpId();
+
+    /**
+     * Consent Management Platform version of the CMP that last updated this TC String
+     *
+     * @since 1.0
+     * @return version of the Consent Management Platform that updated this record
+     */
+    short getCmpVersion();
+
+    /**
+     * 
+     * CMP Screen number at which consent was given for a user with the CMP that last updated this
+     * TC String.
+     * 
+     * The number is a CMP internal designation and is CmpVersion specific. The number is used for
+     * identifying on which screen a user gave consent as a record.
+     *
+     * @since 1.0
+     * @return the screen number identifier
+     */
+    byte getConsentScreen();
+
+    /**
+     * Two-letter ISO 639-1 language code in which the CMP UI was presented.
+     *
+     * @since 1.0
+     * @return the language string
+     */
+    String getConsentLanguage();
+
+    /**
+     * Number corresponds to GVL vendorListVersion. Version of the GVL used to create this TC
+     * String.
+     * 
+     * @since 1.0
+     * @return the version number
+     */
+    short getVendorListVersion();
+
+    /**
+     * The user’s consent value for each Purpose established on the legal basis of consent. The
+     * Purposes are numerically identified and published in the Global Vendor List.
+     * 
+     * @since 1.0
+     * @return The integer values for each established Purpose.
+     */
+    IntIterable getPurposesConsent();
+
+    /**
+     * The user’s consent value for each Purpose established on the legal basis of consent. The
+     * Purposes are numerically identified and published in the Global Vendor List.
+     * 
+     * An alias for PurposesConsent
+     * 
+     * @since 2.0
+     * @return The integer values for each established Purpose.
+     */
+    IntIterable getPurposesAllowed();
```

Comment:
I would not include duplicate methods for the same data even if they come from different versions. This and getPurposesConsent return the same exact data IIRC. We should just have one method.